### PR TITLE
Use language agnostic heuristic to check if monero_wallet_rpc is ready

### LINKED ISSUE #353: Bad line detection when swap start monero_wallet_rpc
The line detection to detect if monero_wallet_rpc has been started in source swap/src/monero/wallet_rpc.rs:150 can be translated in many different language. So the line detection can't work if the local settings on the computer is configured to be other than english like in french for example.

### PR DESCRIPTION
**Pull request by rishflab**
Our strategy of searching for a english string to determine if
monero_wallet_rpc is ready is not compatible with languages other than
english. Instead we assume the monero rpc is ready if it has stopped
writing to stdout. We make a json rpc request to confirm this. A better
solution would have been to configure the monero_wallet_rpc to always
output in english but there is not command line argument to configure
the language.

Closes #353.

NOTE: will squash after approved

### CODE REVIEW

File: swap/src/monero/wallet_rpc.rs [hunk 1]
```diff
@@ -146,12 +148,18 @@ impl WalletRpc {
...
+        // it is is ready
+        while let Ok(res) = tokio::time::timeout(Duration::from_secs(3), reader.next_line()).await {
+            res?;
         }
 
+        // Send a json rpc request to make sure monero_wallet_rpc is ready
+        Client::localhost(port)
+            .get_version()
+            .await
+            .expect("monero wallet rpc is ready to accept json rpc requests");
```
Comment (thomaseizinger) [inline swap/src/monero/wallet_rpc.rs:161]: Why are we panicking here instead of returning a Result?

File: swap/src/monero/wallet_rpc.rs [hunk 2]
```diff
@@ -146,12 +148,18 @@ impl WalletRpc {
 
         let mut reader = BufReader::new(stdout).lines();
 
-        while let Some(line) = reader.next_line().await? {
-            if line.contains("Starting wallet RPC server") {
-                break;
-            }
+        // If we do not hear from the monero_wallet_rpc process for 3 seconds we assume
+        // it is is ready
+        while let Ok(res) = tokio::time::timeout(Duration::from_secs(3), reader.next_line()).await {
```
Comment (thomaseizinger) [inline swap/src/monero/wallet_rpc.rs:153]: I know that technically, `res` is not the read line but the `Result` but can we still call it `line`? Makes it easier to understand I think.

**Review by thomaseizinger — APPROVED** [review]
Fine with me. Out of curiosity: Have you tried figuring out how `monero-wallet-rpc` determines what the current locale is? I think at least on Unix, this happens through the `LANG` env variable so we could just overwrite that when starting it.

**Comment by thomaseizinger** [comment]
Please update the changelog! :)

File: swap/src/monero/wallet_rpc.rs (hunk 2 above)
Comment (rishflab) [reply swap/src/monero/wallet_rpc.rs]: fixed

**Comment by rishflab** [comment]
> Fine with me. Out of curiosity: Have you tried figuring out how `monero-wallet-rpc` determines what the current locale is? I think at least on Unix, this happens through the `LANG` env variable so we could just overwrite that when starting it.

Just tried it and I can change the language by setting the `LANG` env variable. I will have to investigate further to make sure windows works the same way.

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,4 +11,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 - A changelog file.
 
+### Fixed
+
+- Make monero_wallet_rpc readiness check language agnostic. The readiness check was
+  failing on non-english language systems preventing users from starting the swap-cli
+  and asb.
```
Comment (thomaseizinger) [inline CHANGELOG.md:18]: I'd prefer if we stick to the one-sentence-per-line rule for text source files like this.
Git works on a line-by-line basis and in natural language, one usually changes full sentences.

**Comment by thomaseizinger** [comment]
> > Fine with me. Out of curiosity: Have you tried figuring out how `monero-wallet-rpc` determines what the current locale is? I think at least on Unix, this happens through the `LANG` env variable so we could just overwrite that when starting it.
> 
> Just tried it and I can change the language by setting the `LANG` env variable. I am not sure how to test it on windows without merging it and making a release nor could I find anything in the source. Not sure how to proceed, maybe a compile time flag for windows or we can just release it and see if windows users are having the issue?

Yeah not sure if that would work on Windows. We could add an integration test for the whole `wallet_rpc` module maybe? We need to start running tests on Windows in CI then but that might be a good idea anyway.

Once that is in place, you could test it with a commit that sets the language to french for example and if the test then fails on Windows, then you know that it respects the `LANG` variable.

I just checked and `monero-wallet-rpc` starts up fine even if the provided `daemon-host` is garbage (i.e. `localhost` for example) so you can specify that in the tests without need a network connection.

Thinking about it, we would need a network connection anyway for the download so I am not sure if that is such a good idea in the end :sweat_smile:

File: CHANGELOG.md (hunk 3 above)
Comment (rishflab) [reply CHANGELOG.md:18]: We will have to configure dprint to accept it then. Doesn't pass CI unless I `dprint fmt`

Comment (thomaseizinger) [reply CHANGELOG.md:18]: Oh really? :(

My beloved dprint is wrapping markdown files at a certain character range? Damn.

Comment (thomaseizinger) [reply CHANGELOG.md:18]: I guess we could just set the line length to 200 chars or something: https://dprint.dev/plugins/markdown/config/

**Review by thomaseizinger — APPROVED** [review]
I think we can merge this as is :)

**Pull request merged**